analyses/doublet-detection/exploratory-notebooks/03_compare-benchmark-results.Rmd

```diff
@@ -0,0 +1,522 @@
+      dataset_df |>
+        dplyr::rowwise() |>
+        # Add column `consensus_call`
+        #  "doublet" if all three methods are doublet, and "singlet" otherwise
+        dplyr::mutate(
+          consensus_call = dplyr::if_else(
+            all(c(scdbl_prediction, scrublet_prediction, cxds_prediction) == "doublet"),
+            "doublet", 
+            "singlet"
+          )
+        ) |>
+        # Add column `consensus_class`
+        #  "doublet" if all three methods are doublet, "singlet" if all three methods are singlet, 
+        #  and "ambiguous" if there are any disagreements
+        dplyr::mutate(
+          consensus_class = dplyr::case_when(
+            consensus_call == "doublet" ~ "doublet",
+            all(c(scdbl_prediction, scrublet_prediction, cxds_prediction) == "singlet") ~ "singlet", 
+            .default = "ambiguous"
+          )
+         ) |>
+        # Add column `confusion_call`
+        #  tp/tn/fp/fn based on the `consensus_call` column
+        dplyr::mutate(
+          confusion_call = dplyr::case_when(
+            consensus_call == "doublet" && ground_truth == "doublet" ~ "tp",
+            consensus_call == "singlet" && ground_truth == "singlet" ~ "tn",
+            consensus_call == "doublet" && ground_truth == "singlet" ~ "fp",
+            consensus_call == "singlet" && ground_truth == "doublet" ~ "fn"
+          )
+        )
+      }
+  ) 
+```
+      
+
+### PCA
+
+This section plots the PCA for each dataset, clockwise from the top left:
+
+1. `scDblFinder` singlet/doublet calls
+2. `scrublet` singlet/doublet calls
+3. `cxds` singlet/doublet calls 
+4. Ground truth single/doublets
+5. Consensus _call_ droplets are either "doublet" or "singlet"
+6. Consensus _class_: droplets are either "doublet", "singlet", or "ambiguous"
+
+- For the first five panels, doublets are shown in black and singlets in light gray.
+- In the sixth panel showing the consensus class, doublets are shown in black, singlets in light gray, and ambiguous are blue.
+
+```{r, fig.width = 12, fig.height = 6}
+doublet_df_list |>
+  purrr::imap(
+    \(df, dataset) {
+      
+      pc_color_columns <- c("scDblFinder prediction" = "scdbl_prediction",
+                            "scublet prediction" = "scrublet_prediction",
+                            "cxds prediction" = "cxds_prediction",
+                            "Ground truth" = "ground_truth", 
+                            "Consensus call" = "consensus_call")
+      
+      pc_plot_list <- pc_color_columns |> 
+        purrr::imap(
+          \(color_column, plot_title) {
+            plot_pca_calls(
+              df, 
+              color_column = !!sym(color_column), 
+              plot_type = "calls",
+              title = plot_title
+            ) + theme(legend.position = "none")}
+        ) 
+      # plot for consensus_class, which needs three colors
+      # note that the list name isn't actually used here
+      pc_plot_list$consensus_class <- plot_pca_calls(
```

Comment:
Can we add back the tp/fp/tn/fn plot too? I know that means there will be 7 plots but I think that was helpful along with the ambiguous plot.